src/Pool.ts

```diff
@@ -374,13 +465,10 @@ export class Pool<T> {
 
   _validateResource(resource: T) {
     try {
-      return !!this.validate(resource);
+      return Promise.resolve(this.validate(resource));
```

Comment:
you can make function async, and then it will wrap results of awaited function into promise automatically for you.

Reply:
I didn't want to mix async / await in yet in this PR.